clipper_admin/clipper_admin/deployers/mxnet.py

```diff
@@ -0,0 +1,220 @@
+    func : function
+        The prediction function. Any state associated with the function will be
+        captured via closure capture and pickled with Cloudpickle.
+    mxnet_model : mxnet model object
+        The MXNet model to save.
+    mxnet_data_shapes : list(int)
+        List of integers representing the dimensions of data used for model prediction.
+        Required because loading serialized MXNet models involves binding, which requires
+        the shape of the data used to train the model.
+        https://mxnet.incubator.apache.org/api/python/module.html#mxnet.module.BaseModule.bind
+
+        NOTE: Clipper may provide the model with variable size input batches. Because MXNet can't
+              handle variable size input batches, we recommend setting batch size for input data
+              to 1, or dynamically reshaping the model with every prediction based on the current
+              input batch size.
+    base_image : str, optional
+        The base Docker image to build the new model image from. This
+        image should contain all code necessary to run a Clipper model
+        container RPC client.
+    labels : list(str), optional
+        A list of strings annotating the model. These are ignored by Clipper
+        and used purely for user annotations.
+    registry : str, optional
+        The Docker container registry to push the freshly built model to. Note
+        that if you are running Clipper on Kubernetes, this registry must be accesible
+        to the Kubernetes cluster in order to fetch the container from the registry.
+    num_replicas : int, optional
+        The number of replicas of the model to create. The number of replicas
+        for a model can be changed at any time with
+        :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+    Example
+    -------
+
+        from clipper_admin import ClipperConnection, DockerContainerManager
+        from clipper_admin.deployers.mxnet import deploy_mxnet_model
+        import mxnet as mx
+
+        clipper_conn = ClipperConnection(DockerContainerManager())
+
+        # Connect to an already-running Clipper cluster
+        clipper_conn.connect()
+
+        # Create a MXNet model
+        # Configure a two layer neuralnetwork
+        data = mx.symbol.Variable('data')
+        fc1 = mx.symbol.FullyConnected(data, name='fc1', num_hidden=128)
+        act1 = mx.symbol.Activation(fc1, name='relu1', act_type='relu')
+        fc2 = mx.symbol.FullyConnected(act1, name='fc2', num_hidden=10)
+        softmax = mx.symbol.SoftmaxOutput(fc2, name='softmax')
+
+        # Initialize the module and fit it
+        mxnet_model = mx.mod.Module(softmax)
+        mxnet_model.fit(data_iter, num_epoch=1)
+
+        deploy_mxnet_model(
+            clipper_conn,
+            name="example",
+            version = 1,
+            input_type="doubles",
+            func=predict,
+            mxnet_model=model,
+            mxnet_data_shapes=data_shape)
```

Comment:
Where is the `data_shape` variable defined in this example?

Reply:
Oops, forgot to define. Just added in.